python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py

```diff
@@ -885,80 +893,53 @@ def _fetch_and_attach_row_count_metadata(
-                        {table_str}
-                    """,
-                    fetch=True,
-                )
-            row_count = query_result[1][0]["row_count"]
-            additional_metadata = {**TableMetadataSet(row_count=row_count)}
+            # Avoid counting rows for views, since they may include complex SQL queries
+            # that are costly to execute. We can revisit this in the future if there is
+            # a demand for it.
+            if is_view:
+                return event
 
-            if isinstance(event, Output):
-                return event.with_metadata(metadata={**event.metadata, **additional_metadata})
-            else:
-                return event._replace(metadata={**event.metadata, **additional_metadata})
-        except Exception as e:
-            logger.exception(
-                f"An error occurred while fetching row count for {unique_id}. Row count metadata"
-                " will not be included in the event.\n\n"
-                f"Exception: {e}"
-            )
-            return event
+            unique_id = dbt_resource_props["unique_id"]
+            logger.debug("Fetching row count for %s", unique_id)
+            table_str = f"{dbt_resource_props['database']}.{dbt_resource_props['schema']}.{dbt_resource_props['name']}"
 
-    def _stream_dbt_events_and_enqueue_postprocessing(
-        self,
-        output_events_and_futures: List[Union[Future, DbtDagsterEventType]],
-        executor: ThreadPoolExecutor,
-    ) -> None:
-        """Task which streams dbt events and either directly places them in
-        the output_events list to be emitted to user code, or enqueues post-processing tasks
-        where needed.
-        """
-        for event in self:
-            # For any materialization or output event, we run postprocessing steps
-            # to attach additional metadata to the event.
-            if isinstance(event, (AssetMaterialization, Output)):
-                output_events_and_futures.append(
-                    executor.submit(
-                        self._fetch_and_attach_row_count_metadata,
-                        event,
+            try:
+                with adapter.connection_named(f"row_count_{unique_id}"):
+                    query_result = adapter.execute(
+                        f"""
+                            SELECT
+                            count(*) as row_count
+                            FROM
+                            {table_str}
+                        """,
+                        fetch=True,
                     )
+                row_count = query_result[1][0]["row_count"]
+
+                new_metadata = {
+                    **TableMetadataSet(row_count=row_count),
+                }
+                if isinstance(event, Output):
+                    return event.with_metadata({**event.metadata, **new_metadata})
+                else:
+                    return event._replace(metadata={**event.metadata, **new_metadata})
```

Comment:
Prob should rebase, since I believe you landed a PR that consolidates this behavior